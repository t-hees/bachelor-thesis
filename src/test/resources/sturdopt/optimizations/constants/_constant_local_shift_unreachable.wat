(module
  (type $func_type (func (param i32) (result i32)))
  (memory $memory 1)
  (func $func (type $func_type) (local $loc1 i32)
    local.get 0       ;; i32.const 0
    if (result i32)
      local.get 0
    else
      i32.const 0
    end
    i32.load
    local.set 1      ;; needs to be shifted after opt
    local.get 1
  )
  (func (export "main") (type $func_type)
    i32.const 0
    local.get 0
    i32.store

    i32.const 0
    call $func
  )
)
