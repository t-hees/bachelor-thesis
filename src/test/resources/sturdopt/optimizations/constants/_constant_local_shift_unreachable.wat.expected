(module
  (type $func_type (func (param i32) (result i32)))
  (memory $memory 1)
  (func $func (result i32) (local $loc1 i32)
    i32.const 0
    if (result i32)
      unreachable
    else
      i32.const 0
    end
    i32.load
    local.set 0 ;; shifted
    local.get 0
  )
  (func (export "main") (type $func_type)
    i32.const 0
    local.get 0
    i32.store

    i32.const 0
    drop
    call $func
  )
)
